addons/payment_adyen/models/payment_provider.py

```diff
@@ -51,15 +48,18 @@ def write(self, values):
         return super().write(values)
 
     @api.model
-    def _adyen_trim_api_urls(self, values):
+    def _adyen_trim_api_urls(self, values): #here I just want to strip it to only prefix
         """ Remove the version and the endpoint from the url of Adyen API fields.
 
         :param dict values: The create or write values
         :return: None
         """
-        for field_name in ('adyen_checkout_api_url', 'adyen_recurring_api_url'):
-            if values.get(field_name):  # Test the value in case we're duplicating a provider
-                values[field_name] = re.sub(r'[vV]\d+(/.*)?', '', values[field_name])
+
+        if values.get('adyen_api_url_prefix'):  # Test the value in case we're duplicating a provider
+            if values.get('state') == 'test':
+                values['adyen_api_url_prefix'] = 'test'
+            else:
+                values['adyen_api_url_prefix'] = re.sub(r'(https:\/\/)*(\w+-\w+)(-.*|\..*)*', r'\2', values['adyen_api_url_prefix'])
```

Comment:
```suggestion
                values['adyen_api_url_prefix'] = re.sub(r'(?:https://)?(\w+-\w+).*', r'\1', values['adyen_api_url_prefix'])
```

Reply:
As per PM discussion, my understanding is that this REGEX should take this url: `https://whatever-odoosa1234-checkout-test.com/checkout/` and match only `whatever-odoosa1234`. In other words we want to match the first two words (composed of letters, numbers and underscores) separated by an hyphen `-`.

In order to do so I propose to simplify this regex a bit:

1. Using `?:` to declare that first group doesn't need to be captured
2. Removing the escaping for the `/` chars, which is not a special character in regex, thus does not need to be escaped (I might be missing something here)
3. Changing `*` in front of the domain with `?` (unless we also want to match strings like: `https://https://test-try-...`)
4. Just matching anything (`.*`) after the capture group. Unless there are some specific patterns that want to be preserved (e.g. `https://foo-bar/baz` would return `foo-bar/baz`, since the `/` is not one of the specified options).

One remark which I think is worth making is that this won't work as expected (not removing the first part of the string) on `https://www.whatever-odoosa1234-checkout-test.com/checkout/`, since the `www.` is not accounted for. In order to solve this (if that case could ever happen) I'd propose the following instead: '(?:https://)?.*?(\w+-\w+).*'.